### PR TITLE
Fixes from internal testing

### LINKED ISSUE #244: Rename PretokenizationPipeline and PosttokenizationPipeline
PosttokenizationPipeline and PretokenizationPipeline sounds too similar to pipeline.py module and should be renamed to be more related to hooks. 

Suggestions: 
- PretokenizationHookManager / PosttokenizationHookManager
- PretokenizationQueue / PosttokenizationQueue

Further, hook code could be extracted to a separate module and unified using a general HookQueue interface that defines `add_hook` and `clear` methods to avoid copying of those.

### PR DESCRIPTION
**Pull request by mttk**
**Functionality**
* `finalize_fields` is something the user now has to call to construct the Vocab
  * Issues with huggingface/SST
  * More explicit control what the vocab is built over
* Eager now defaults to False
  * Issues with huggingface/SST would require setting it to False anyway
  * Better that people know the issues with setting it to true (the frequencies will be built based on all datasets loaded prior to finalization)
* ~~Print format for Examples changed: one Field content per line~~
* Added an option to not use padding directly in `SingleBatchIterator` (the new `add_padding` argument)
  * To make this easier, `Iterator` now has a new `disable_batch_matrix` argument to disable matrix batching on the level of the entire dataset

**Docs**
* Added a quickstart section -- minimal text overview of usage
* Major docstring visual revamp & text trimming
* Changed ``http`` to ``https`` in takelab urls and updated repository references
* Deleted FAQ until populated
* Reordered some sections in walkthrough, added vocab finalization notes

**Misc**
* Bumped down version to 0.1.0
* Updated placeholder logo

Closes #244

### CODE REVIEW

File: podium/datasets/dataset.py [hunk 1]
```diff
@@ -165,7 +181,39 @@ def batch(self) -> Tuple[NamedTuple, NamedTuple]:
         # Imported here because of circular import
         from podium.datasets import SingleBatchIterator
 
-        return next(iter(SingleBatchIterator(self, shuffle=False)))
+        return next(
+            iter(SingleBatchIterator(self, shuffle=False, add_padding=add_padding))
+        )
+
+    def as_dict(self, include_raw=False):
```
Comment (FilipBolt) [inline podium/datasets/dataset.py:188]: Can you add the typing spec for return type?

**Review by FilipBolt — APPROVED** [review]
LGTM

**Review by mariosasko — APPROVED** [review]
LGTM!

File: podium/datasets/dataset.py [hunk 2]
```diff
@@ -165,7 +181,39 @@ def batch(self) -> Tuple[NamedTuple, NamedTuple]:
...
+        """
+        dataset_as_dict = {}
+
+        for field in self.fields:
+            field_values = getattr(self, field.name)
+            if not include_raw:
+                # Strip raw data
+                _, field_values = zip(*field_values)
+
+            dataset_as_dict[field.name] = list(field_values)
```
[the remark this answers is not in the document]
Comment (mttk) [reply podium/datasets/dataset.py:214]: Cast to list is to copy the data and make it concrete, and as it is intended people use this data elsewhere I'm not sure we need to care for immutability

**Review by ivansmokovic — APPROVED** [review]
LGTM

**Pull request merged**